src/Sarif/Writers/PrereleaseCompatibilityTransformer.cs

```diff
@@ -1716,5 +1684,85 @@ private static bool ConvertExceptionMessageFromStringToMessageObject(JObject exc
 
             return modifiedNotification;
         }
+
+        private delegate void ActionOnJObject(JObject jObject);
+
+        private static void ApplyActionOnNodeIfPathExists(string[] possibleNodePaths, JObject rootNode, ActionOnJObject action)
+        {
+            var nodesToUpdate = FindAllNodes(possibleNodePaths, rootNode);            
+
+            foreach (JObject node in nodesToUpdate)
+            {
+                action(node);
+            }
+
+        }
+
+        private static List<JObject> FindAllNodes(string[] fullPaths, JObject rootlog)
+        {
+            var nodes = new List<JObject>();
+
+            foreach (string path in fullPaths)
+            {
+                nodes.AddRange(GetAllNodesAtPath(path, rootlog));
+            }
+
+            return nodes;
+        }
+
+        private static List<JObject> GetAllNodesAtPath(string fullPath, JObject rootLog)
+        {
+            if (fullPath == null)
+            {
+                return new List<JObject>() { rootLog };
+            }
+
+            (string currentNodeName, string childPath) = SplitCurrentNodeNameAndChildPath(fullPath);
+
+            var leafNodes = new List<JObject>();
+
+            if (currentNodeName.EndsWith("[]"))
+            {
+                currentNodeName = currentNodeName.TrimEnd("[]".ToCharArray());
+
+                if (rootLog[currentNodeName] is JArray currentLogs)
+                {
+                    foreach (JObject currentLog in currentLogs)
+                    {
+                        leafNodes.AddRange(GetAllNodesAtPath(childPath, currentLog));
+                    }
+                }
+            }
+            else
+            {
+                if (rootLog[currentNodeName] is JObject currentLog)
+                {
+                    leafNodes.AddRange(GetAllNodesAtPath(childPath, currentLog));
+                }
+            }
+
+            return leafNodes;
+
+        }
+
+        private static (string, string) SplitCurrentNodeNameAndChildPath(string fullPath)
+        {
+            if (string.IsNullOrWhiteSpace(fullPath))
+            {
+                throw new ArgumentNullException(nameof(fullPath));
+            }
+
+            char[] delimiter = { '.' };
+
+            string[] splittedItems = fullPath.Split(delimiter, 2);
+
+            if(splittedItems.Length == 1)
+            {
+                return (splittedItems[0], null);
+            }
+
+            return (splittedItems[0], splittedItems[1]);
```

Comment:
>splittedItems[0] [](start = 20, length = 16)

need better naming for readability

Reply:
with named params, i think its clearer, having local vars defined with the same name seems overkill (i tried it but hated it). This version looks readable to me :)

---
In reply to: [266093407](https://github.com/Microsoft/sarif-sdk/pull/1349#discussion_r266093407) [](ancestors = 266093407)